tfsdk/serve.go

```diff
@@ -38,12 +38,39 @@ type ServeOpts struct {
 
 // NewProtocol6Server returns a tfprotov6.ProviderServer implementation based
 // on the passed Provider implementation.
+//
+// Deprecated: Use NewProtocol6ProviderServer instead. This will be removed
+// in an upcoming minor version before v1.0.0.
```

Comment:
I was going to ask if we really need to do this deprecation phase, but I did a quick GitHub global code search, and while it's in the order of ~70 usages, and it includes the HashiCorp repos, it's still 70 usages.

Yep, deprecation first it is.